### PR TITLE
expand to py310 & py311

### LINKED ISSUE #159: How does libhkl get wavelength for add_reflection()?
https://github.com/bluesky/hklpy/blob/19b397aa9244af1d8340abab321e7395e1310feb/hkl/user.py#L205

### LINKED ISSUE #259: Expand support to include Py3.10 & 3.11
Once the [libhkl v5.0.0.3001](https://github.com/conda-forge/hkl-feedstock/pull/13) package is ready on conda-forge, expand the CI testing to Python 3.10 & 3.11 (Py3.8-3.11).  If successful, push a new release (v1.0.4).

- https://github.com/conda-forge/hkl-feedstock/pull/13

### PR DESCRIPTION
**Pull request by prjemian**
- close #259

### CODE REVIEW

**Comment by prjemian** [comment]
[`bluesky.simulation.check_limits()`](https://blueskyproject.io/bluesky/generated/bluesky.simulators.check_limits.html#bluesky.simulators.check_limits) was refactored and can't be used here since it errors with:

```python
RuntimeError: Bluesky event loop not running
```

Need to refactor some tests.

**Comment by prjemian** [comment]
Reducing workflow duplication in [0411e9f](https://github.com/bluesky/hklpy/pull/261/commits/0411e9f065ddadb1bdf0a432af38d902edf1ccb7).

**Comment by prjemian** [comment]
@mrakitin , @klauer : This is ready for review.

File: .github/workflows/conda_unit_test.yml [hunk 1]
```diff
@@ -1,24 +1,76 @@
...
+          pip install flake8
+
+      - name: Run flake8
+        run: |
+          flake8
+
+      - name: Run black
+        uses: rickstaa/action-black@v1
+        with:
+          black_args: ". --check"
```
Comment (mrakitin) [inline .github/workflows/conda_unit_test.yml:42]: Maybe all these checks can be done via pre-commit as we started doing for other projects?
https://github.com/NSLS-II/scientific-python-cookiecutter/blob/main/%7B%7B%20cookiecutter.repo_name%20%7D%7D/.github/workflows/pre-commit.yml

File: .github/workflows/conda_unit_test.yml [hunk 2]
```diff
@@ -1,24 +1,76 @@
...
     runs-on: ubuntu-latest
+    needs: lint
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version:
+        - 3.8
+        - 3.9
+        - "3.10"
+        - 3.11
```
Comment (mrakitin) [inline .github/workflows/conda_unit_test.yml:54]: I'd convert all of them to strings for consistency.
```suggestion
        - "3.8"
        - "3.9"
        - "3.10"
        - "3.11"
```

File: .github/workflows/conda_unit_test.yml [hunk 3]
```diff
@@ -1,24 +1,76 @@
...
+    shell: bash -l {0}
 
 jobs:
 
+  lint:
+    name: Code style
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (mrakitin) [inline .github/workflows/conda_unit_test.yml:24]: Version 4 is the latest now.
```suggestion
      - uses: actions/checkout@v4
```

File: hkl/tests/test_util.py [hunk 4]
```diff
@@ -63,7 +63,7 @@ def test_software_versions_default_list(case):
         ("hkl", "5.0.0"),
         ("hklpy", "0"),  # minimum test for unversioned use
         (NO_SUCH_PACKAGE_NAME, "---"),
-        ("ophyd", "1.6"),
+        # ("ophyd", "1.6"),  conda has right version but pip has 0.0.0
```
Comment (mrakitin) [inline hkl/tests/test_util.py:66]: @evalott100, is it something that can be easily fixed in the `pyproject.toml` config for ophyd?

**Review by mrakitin — APPROVED** [review]
Looks good to me, @prjemian!

I made a few suggestions mostly regarding the CI/pre-commit. They can be addressed after the release.

File: .github/workflows/conda_unit_test.yml (hunk 1 above)
Comment (prjemian) [reply .github/workflows/conda_unit_test.yml:42]: Good idea. Added to next release.

**Review by klauer — COMMENTED** [review]
Late review as I'm at a conference but this LGTM :+1:

**Pull request merged**